pkg/fqdn/helpers.go

```diff
@@ -36,14 +37,6 @@ func (n *NameManager) MapSelectorsToIPsLocked(fqdnSelectors map[api.FQDNSelector
 			dnsName := prepareMatchName(ToFQDN.MatchName)
 			lookupIPs := n.cache.Lookup(dnsName)
 
-			// Mark this FQDNSelector as having no IPs corresponding to it.
-			// FQDNSelectors are guaranteed to have only their MatchName OR
-			// their MatchPattern set (having both set is invalid per
-			// sanitization of FQDNSelectors).
-			if len(lookupIPs) == 0 {
-				missing[ToFQDN] = struct{}{}
-			}
```

Comment:
I couldn't quite tell at a glance, but what's the reason why the new implementation no longer needs to know which selectors lack IPs?